x/gamm/pool-models/stableswap/pool.go

```diff
@@ -418,13 +418,22 @@ func validateScalingFactors(scalingFactors []uint64, numAssets int) error {
 
 // assumes liquidity is all pool liquidity, in correct sorted order
 func validatePoolLiquidity(liquidity sdk.Coins, scalingFactors []uint64) error {
+	liquidityCopy := make(sdk.Coins, len(liquidity))
```

Comment:
Any reason `sdk.NewCoins(liquidity...)` would work?

Reply:
agreed, that seems better to me as well

Reply:
It also sorts the input coins. We don't want that

As a result, we would have to make a new copy anyways